setup-dev-env.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+if [ -f apps/backend/.env ]; then
+	echo "apps/backend/.env already exists, if you would like to regenerate your secrets, please delete this file and re-run the script."
+else
+	echo "apps/backend/.env does not exist, creating..."
+	(umask 066; touch apps/backend/.env)
+fi
+
+if ! grep -qF "NODE_ENV" apps/backend/.env; then
+	echo "NODE_ENV=development" >> apps/backend/.env
+fi
+
+if ! grep -qF "DATABASE_USERNAME" apps/backend/.env; then
+  	read -p "Enter DATABASE_USERNAME (leave blank to use default 'postgres': " usr
+	echo "DATABASE_USERNAME=$usr" >> apps/backend/.env
+fi
+
+if ! grep -qF "DATABASE_PASSWORD" apps/backend/.env; then
+  	read -p 'Enter DATABASE_PASSWORD (leave blank to not set a password): ' psswd
+	echo "DATABASE_PASSWORD=$psswd" >> apps/backend/.env
+fi
+
+if ! grep -qF "JWT_EXPIRE_TIME" apps/backend/.env; then
+  	read -p 'Enter JWT_EXPIRE_TIME ex. 1d or 25m: ' expire
+	echo "JWT_EXPIRE_TIME=$expire" >> apps/backend/.env
+fi
+
+if ! grep -qF "JWT_SECRET" apps/backend/.env; then
+	echo "JWT_SECRET=$(openssl rand -hex 64)" >> apps/backend/.env
+fi
+
+if ! grep -qF "API_KEY_SECRET" apps/backend/.env; then
+	read -p "apps/backend/.env does not contain API_KEY_SECRET, would you like to enable API Keys? [Y/n]" -n 1 -r
+	if [[ $REPLY =~ ^[Yy]$ ]]
+	then
+		echo "API_KEY_SECRET=$(openssl rand -hex 33)" >> apps/backend/.env
+	fi
+fi
+
+if ! grep -qF "NGINX_HOST" apps/backend/.env; then
+	echo "apps/backend/.env does not contain NGINX_HOST..."
+	read -p 'Enter your FQDN/Hostname/IP Address: ' fqdn
+	echo "NGINX_HOST=$fqdn" >> apps/backend/.env
+fi
+
+if [ -f ./nginx/certs/ssl_certificate.crt ]; then
+	echo "SSL Certificate already exists. if you would like to regenerate your certificates, please delete the files in ./nginx/certs/ and re-run this script."
+else
+	echo "SSL Certificate does not exist, creating self-signed certificate..."
+	echo "Do not use a self-signed certificate in a production environment."
```

Comment:
I wouldn't necessarily say this though it's a good warning usually.  Self-signed certs could for example be used if you're deployed behind a bastion of some kind which includes authentication/authorization processes so that anything behind it is actually good to go and all you need is encryption for data in transit.  Moreover, companies often introduce self-signed certs when they'd like to act as middlemen and inspect their own traffic.  Just means that you need to explicitly add it to your list of trusted certs.  Kinda wandering here but yeah... I would make this more of a warning than a prohibition.

https://sectigostore.com/page/what-is-a-self-signed-certificate  Not a bad article just ignore the self-advertisements